dan_layer/engine/tests/templates/nft/emoji_id/src/lib.rs

```diff
@@ -84,16 +84,15 @@ mod emoji_id {
             }
         }
 
-        // TODO: return change
+        // TODO: return change (or check bucket.value() == required_amount)
```

Comment:
I think a check like this is all that is needed
`assert_eq!(bucket.amount(), required_amount, "Not enough funds transferred for payment")`